### ISSUE TITLE
Error: Build Elixir From Source (Elixir 1.9.4)

### ISSUE THREAD

**Issue by meetlost** [elixir-lang/elixir#9693]
### Environment

* Elixir & Erlang/OTP versions (elixir --version): OTP-22.1.4 Elixir 1.9.4
* Operating system: FreeBSD 10.3-RELEASE
* CPU:  0.0% user,  0.0% nice,  0.0% system,  0.0% interrupt,  100% idle
* Mem: 352M Active, 63M Inact, 200M Wired, 1432K Cache, 109M Buf, 357M Free
* Erlang Version (detail): Erlang/OTP 22 [erts-10.5.3] [source] [64-bit] [smp:2:2] [ds:2:2:10] [async-threads:1] [hipe]

### Current behavior

==> elixir (compile)
make[1]: Entering directory '/usr/home/by/projects/elixir'
==> unicode (compile)
Compiling /usr/home/by/projects/elixir/lib/elixir/unicode/unicode.ex (it's taking more than 15s)
make[1]: *** [Makefile:105: lib/elixir/ebin/Elixir.String.Unicode.beam] Killed
make[1]: Leaving directory '/usr/home/by/projects/elixir'
make: *** [Makefile:92: lib/elixir/ebin/Elixir.Kernel.beam] Error 2

### Expected behavior

Compile ok.

### Remark
I searched it with Google, and got below info (issue): https://github.com/elixir-lang/elixir/issues/7319

If memory shortage is the cause, how can I know how much memory do I need for the compile?

**Comment by josevalim** [comment]
Unfortunately there is no pre-determined amount of memory besides checking out previous reports and giving them a try. 1GB should be enough, please let us know how it goes.

**Comment by meetlost** [comment]
I am not sure how to test this with the most accurate method.

First, I try to use ulimit [-v|-m] to constrain the memory, but the result is strange. Even when I set the maximum memory limit to 3GB, the compilation failed with the same error (4GB is ok). If I set the memory and swap space (by default) to 1024MB (using guest machine via VirtualBox), the compilation works. So, 2GB should be ok.

I want more accurate answer.

I disabled the swap space, using 1024MB of physical memory, the compilation failed, so 1024MB is not enough in this case. After some tests, I found that 1300MB is ok, but 1200MB is not.

The more accurate answer I wanted is 1300MB.

All of the tests mentioned above is under FreeBSD 12.0-RELEASE. Since different operating system has different memory management facility, this number might not work on other scenarios. But I think, mostly, it should work.
